src/esphomelib/light/neo_pixel_bus_light_output.h

```diff
@@ -0,0 +1,216 @@
+    this->schedule_show();
+  }
+
+  void setup() override {
+    ESP_LOGCONFIG(TAG, "Setting up Neo Pixel Bus light...");
+    assert(this->controller_ != nullptr && "You need to add LEDs to this controller!");
+    this->controller_->ClearTo(typename T_COLOR_FEATURE::ColorObject(0, 0, 0));
+  }
+  void loop() override {
+    if (!this->next_show_)
+      return;
+
+    uint32_t now = micros();
+    this->last_refresh_ = now;
+    this->next_show_ = false;
+
+    ESP_LOGVV(TAG, "Writing RGB values to bus...");
+
+#ifdef USE_OUTPUT
+    if (this->power_supply_ != nullptr) {
+      bool is_light_on = false;
+      for (int i = 0; i < this->controller_->PixelCount(); i++) {
+        auto color = this->controller_->GetPixelColor(i);
+        if (is_on(color)) {
+          is_light_on = true;
+          break;
+        }
+      }
+
+      if (is_light_on && !this->has_requested_high_power_) {
+        this->power_supply_->request_high_power();
+        this->has_requested_high_power_ = true;
+      }
+      if (!is_light_on && this->has_requested_high_power_) {
+        this->power_supply_->unrequest_high_power();
+        this->has_requested_high_power_ = false;
+      }
+    }
+#endif
+
+    this->controller_->Show();
+  }
+  float get_setup_priority() const override {
+    return setup_priority::HARDWARE;
+  }
+  template <typename U>
+  static typename std::enable_if<std::is_same<U, RgbColor>::value, RgbColor>::type get_light_color(
+      LightState *state, const LightColorValues values) {
+    float red, green, blue;
+    values.as_rgb(&red, &green, &blue);
+    red = gamma_correct(red, state->get_gamma_correct());
+    green = gamma_correct(green, state->get_gamma_correct());
+    blue = gamma_correct(blue, state->get_gamma_correct());
+    uint8_t redb = red * 255;
+    uint8_t greenb = green * 255;
+    uint8_t blueb = blue * 255;
+    return RgbColor(redb, greenb, blueb);
+  }
+
+  template <typename U>
+  static typename std::enable_if<std::is_same<U, RgbwColor>::value, RgbwColor>::type get_light_color(
+      LightState *state, const LightColorValues values) {
+    float red, green, blue, white, brightness;
+    values.as_rgbw(&red, &green, &blue, &white);
+    values.as_brightness(&brightness);
+    red = gamma_correct(red, state->get_gamma_correct());
+    green = gamma_correct(green, state->get_gamma_correct());
+    blue = gamma_correct(blue, state->get_gamma_correct());
+    white = gamma_correct(white, state->get_gamma_correct());
+    uint8_t redb = red * 255;
+    uint8_t greenb = green * 255;
+    uint8_t blueb = blue * 255;
+    uint8_t whiteb = white * 255;
+    uint8_t brightnessb = brightness * 255;
+    // currently in hass there is no way to only show white via the hass ui, so disable the colors if all of them are on
+    // and therefore very white
+    bool white_colors = redb >= brightnessb - 10 && greenb >= brightnessb - 10 && blueb >= brightnessb - 10 &&
+                        whiteb >= brightnessb - 10;

```

Comment:
I see your point and have removed it for now. I'd like to have the functionality as it's a power saving feature. I guess LightState could implement it. What do you think where this would best fit?